tools/Customization/DevHome.Customization/Models/SourceControlIntegration.cs

```diff
@@ -13,21 +14,22 @@ namespace DevHome.Customization.Models;
 
 public class SourceControlIntegration
 {
-    private static readonly Serilog.ILogger Log = Serilog.Log.ForContext("SourceContext", nameof(Models.SourceControlIntegration));
+    private static readonly Serilog.ILogger _log = Serilog.Log.ForContext("SourceContext", nameof(Models.SourceControlIntegration));
+    private static readonly StringResource _stringResource = new("DevHome.Customization.pri", "DevHome.Customization/Resources");
 
     public static SourceControlValidationResult ValidateSourceControlExtension(string extensionCLSID, string rootPath)
     {
         var providerPtr = IntPtr.Zero;
         try
         {
-            Log.Information("Validating source control extension with arguments: extensionCLSID = {extensionCLSID}, rootPath = {rootPath}", extensionCLSID, rootPath);
+            _log.Information("Validating source control extension with arguments: extensionCLSID = {extensionCLSID}, rootPath = {rootPath}", extensionCLSID, rootPath);
 
             var hr = PInvoke.CoCreateInstance(Guid.Parse(extensionCLSID), null, Windows.Win32.System.Com.CLSCTX.CLSCTX_LOCAL_SERVER, typeof(ILocalRepositoryProvider).GUID, out var extensionObj);
             providerPtr = Marshal.GetIUnknownForObject(extensionObj);
             if (hr < 0)
             {
-                Log.Error(hr.ToString(), "Failure occurred while creating instance of repository provider");
-                return new SourceControlValidationResult(ResultType.Failure, ErrorType.RepositoryProvderCreationFailed, null, null, null);
+                _log.Error(hr.ToString(), "Failure occurred while creating instance of repository provider");
+                return new SourceControlValidationResult(ResultType.Failure, ErrorType.RepositoryProviderCreationFailed, null, _stringResource.GetLocalized("ValidateSourceControlErrorOnRepositoryProviderInstanceCreation"), _stringResource.GetLocalized("ValidateSourceControlErrorOnRepositoryProviderInstanceCreation"));
```

Comment:
DiagnosticText shouldn't be explicitly localized. If we don't have anything more internally useful to say in DiagnosticText, I think it's fine to leave null.

Reply:
Yes, this is returned to Dev Home.